### PR TITLE
#152: Allow dots in release candidate suffix

### LINKED ISSUE #152: Allow dots in release candidate suffix
## Situation

Most projects use a dash to separate the release candidate suffix from the rest of the version number.
Unfortunately `docker-db` uses a dot. Example: 7.1.rc1

## Acceptance Criteria

1. Version parser recognizes suffix starting with a dot.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #152.

### CODE REVIEW

File: doc/changes/changes_4.0.1.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+Code name: Allow dots in release candidate suffix
+
+## Summary
+
+Release 4.0.1 contains a small update that allows using dots as separators for release candidate versions.
+
+## Breaking Changes
+
+* `bucket-fs-java` was updated to 2.0.1, which changed the method signature or some methods in order to offer better exceptions.
```
Comment (jakobbraun) [inline doc/changes/changes_4.0.1.md:11]: I can't see an update in the pom.xml

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: doc/changes/changes_4.0.1.md (hunk 1 above)
Comment (redcatbear) [reply doc/changes/changes_4.0.1.md]: Removed.

**Review by jakobbraun — APPROVED** [review]

**Comment by jakobbraun** [comment]
Fix code smell

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -28,21 +28,32 @@ jobs:
...
+          -Dcom.exasol.containers.monitored_exit=EXIT_ANY \
           -Pslow
         env:
           DOCKER_DB_VERSION: ${{ matrix.docker_db_version }}
           GITHUB_OAUTH: ${{ secrets.GITHUB_TOKEN }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
\ No newline at end of file
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SUPPORT_DIR: ${{ github.workspace }}/target/support
```
Comment (jakobbraun) [inline .github/workflows/ci-build.yml:54]: I think  you can directly use this expression in the above command. No need for an env var

File: doc/changes/changelog.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # Changes
 
+* [4.0.0](changes_4.0.1.md)
```
Comment (jakobbraun) [inline doc/changes/changelog.md:3]:
```suggestion
* [4.0.1](changes_4.0.1.md)
```

File: doc/changes/changes_4.0.1.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+## Features
+
+* #152: Version parser now accepts suffixes starting with a dot
+
+## Dependency Updates
+
+### Compile Dependency Updates
+
+* Updated `com.exasol:bucketfs-java:2.0.1` to `2.1.0`
```
Comment (jakobbraun) [inline doc/changes/changes_4.0.1.md:19]: did you update all dependencies?

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml (hunk 2 above)
Comment (redcatbear) [reply .github/workflows/ci-build.yml:54]: Possibly. But I find it cleaner to keep GitHub artifacts out of the shell script part if possible.

File: doc/changes/changes_4.0.1.md (hunk 4 above)
Comment (redcatbear) [reply doc/changes/changes_4.0.1.md:19]: Except for the JDBC driver for which now a 7.1 version is also available. But I want to do 7.1 compatibility separate. A small smoke test showed that there is probably a loot to do for 7.1

**Review by jakobbraun — DISMISSED** [review]

**Pull request merged**